chainer/optimizer.py

```diff
@@ -750,16 +806,18 @@ def update(self, lossfun=None, *args, **kwds):
             del loss
 
         self.reallocate_cleared_grads()
-
+        self.check_nan_in_grads()
```

Comment:
We can make it even less stateful by returning `self._loss_scaling_isnan` from this function instead of having it as an attribute. It can be passed to `update_loss_scale` and we then we no longer need the `is_safe_to_update` method. What do you say?

Reply:
But maybe it's difficult considering the fact that `self_loss_scaling_isnan_ever` is very stateful.